### PR TITLE
Document new fly.toml deploy timeouts

### PR DESCRIPTION
**Pull request by dangra**
### Related Fly.io community and GitHub links
https://github.com/superfly/flyctl/pull/2961

### Notes
Feedback, fixes and rewording much welcome 🤟

### CODE REVIEW

File: reference/configuration.html.markerb [hunk 1]
```diff
@@ -178,6 +178,8 @@ To ensure the command runs in a specific region, such as `dfw`, set `PRIMARY_REG
 
 The environment variable `RELEASE_COMMAND=1` is set within the temporary release VM. You can use this to define behavior in your Dockerfile's `ENTRYPOINT`  that's conditional on being run on a release VM.
 
+By default, the release command has 5 minutes to complete before it is killed. It is possible to increase or decrease the timeout with `fly deploy --release-command-timeout=10m` or by setting `release_command_timeout = "10m"` in `[deploy]` section of `fly.toml`.
```
Comment (andie787) [inline reference/configuration.html.markerb:181]: Q: is there a maximum?

File: reference/configuration.html.markerb [hunk 2]
```diff
@@ -235,6 +237,17 @@ Secrets take precedence over env variables with the same name.
 
 **Note:** In Apps V2, the [`primary_region` option](#primary-region) sets the `PRIMARY_REGION` environment variable within a Machine and overrides any value set in the `[env]` section.
 
+#### Dealing with timeout errors while deploying a new version
+
+Everytime `fly deploy` runs, it builds a new docker image, push it to an internal registry and only then goes to update or create machines for the app. Sometimes the image is too large, or the platform is slower than usual pulling the new image layers, triggering the default 2 minutes wait for machine started timeout.
```
Comment (andie787) [inline reference/configuration.html.markerb:242]:
```suggestion
Every time `fly deploy` runs, it builds a new docker image, pushes it to an internal registry, and then updates or creates machines for the app. Sometimes the image is too large, or the platform is slower than usual pulling the new image layers, triggering the default 2 minute timeout waiting for the Machine to be in a started state.
```
some minor suggestions :)

File: reference/configuration.html.markerb [hunk 3]
```diff
@@ -235,6 +237,17 @@ Secrets take precedence over env variables with the same name.
...
+
+The good thing is that this timeout can be controlled in two ways, as a command line flag `--wait-timeout`. i.e.: `fly deplopy --wait-timeout=10m` or as a more permanent fly.toml setting:
```
Comment (andie787) [inline reference/configuration.html.markerb:244]:
```suggestion
The good thing is that this timeout can be controlled in two ways, with the `--wait-timeout` option, for example `fly deplopy --wait-timeout=10m`, or as a more permanent `fly.toml` setting:
```
a few terminology/style things

**Review by andie787 — COMMENTED** [review]
thanks for this! left a question and some nitpicks

File: reference/configuration.html.markerb (hunk 1 above)
Comment (dangra) [reply reference/configuration.html.markerb:181]: nope

**Comment by andie787** [comment]
super-helpful! 🚢 it when you're ready!

File: reference/configuration.html.markerb [hunk 4]
```diff
@@ -235,6 +237,17 @@ Secrets take precedence over env variables with the same name.
 
 **Note:** In Apps V2, the [`primary_region` option](#primary-region) sets the `PRIMARY_REGION` environment variable within a Machine and overrides any value set in the `[env]` section.
 
+#### Dealing with timeout errors while deploying a new version
+
+Every time `fly deploy` runs, it builds a new docker image, pushes it to an internal registry, and then updates or creates machines for the app. Sometimes the image is too large, or the platform is slower than usual pulling the new image layers, triggering the default 2 minute timeout waiting for the Machine to be in a started state.
```
[the remark this answers is not in the document]
Comment (catflydotio) [reply reference/configuration.html.markerb:242]: We should also get a rule for "Docker"

**Pull request merged**